src/geos_operations.jl

```diff
@@ -123,80 +123,84 @@ union(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context(
 
 # # Snap first geometry on to second with given tolerance
-snap(obj1::Geometry, obj2::Geometry, tol::Real, context::GEOSContext = get_global_context()) =
+snap(obj1::Geometry, obj2::Geometry, tol::Real, context::GEOSContext = get_context(obj1,obj2)) =
     geomFromGEOS(snap(obj1.ptr, obj2.ptr, tol, context), context)
 
 # -----
 # Binary predicates
 # -----
 
-disjoint(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
+disjoint(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1,obj2)) =
     disjoint(obj1.ptr, obj2.ptr, context)
-touches(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
+touches(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1,obj2)) =
     touches(obj1.ptr, obj2.ptr, context)
-intersects(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
+intersects(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1,obj2)) =
     intersects(obj1.ptr, obj2.ptr, context)
-crosses(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
+crosses(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1,obj2)) =
     crosses(obj1.ptr, obj2.ptr, context)
-within(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
+within(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1,obj2)) =
     within(obj1.ptr, obj2.ptr, context)
-Base.contains(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
+Base.contains(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1,obj2)) =
     Base.contains(obj1.ptr, obj2.ptr, context)
-overlaps(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
+overlaps(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1,obj2)) =
     overlaps(obj1.ptr, obj2.ptr, context)
-equals(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
-    equals(obj1.ptr, obj2.ptr, context)
-equalsexact(obj1::Geometry, obj2::Geometry, tol::Real, context::GEOSContext = get_global_context()) =
+equalsexact(obj1::Geometry, obj2::Geometry, tol::Real, context::GEOSContext = get_context(obj1,obj2)) =
     equalsexact(obj1.ptr, obj2.ptr, tol, context)
-covers(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
+covers(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1,obj2)) =
     covers(obj1.ptr, obj2.ptr, context)
-coveredby(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_global_context()) =
+coveredby(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1,obj2)) =
     coveredby(obj1.ptr, obj2.ptr, context)
 
+function equals(obj1::Geometry, obj2::Geometry, context::GEOSContext = get_context(obj1))
+    # We don't check context here, as it is probably safe to compare objects from 
+    # different contexts
```

Comment:
Out of curiosity: is this a verified claim?

Reply:
I am not 100% sure. But we have a unit test on main: https://github.com/JuliaGeo/LibGEOS.jl/blob/master/test/test_geos_types.jl#L54